src/snowflake/snowpark/table_function.py

```diff
@@ -130,6 +131,23 @@ def alias(self, *aliases: str) -> "TableFunctionCall":
 
     as_ = alias
 
+    def __str__(self) -> str:
+        return self.name
+
+
+class _ExplodeFunctionCall(TableFunctionCall):
+    """Internal class to identify explode function call as a special instance of TableFunctionCall"""
+
+    def __init__(self, col: ColumnOrName) -> None:
+        super().__init__("flatten", col)
+        if isinstance(col, Column):
+            self.col = col._expression.name
+        else:
+            self.col = quote_name(col)
+
+    def __str__(self) -> str:
+        return "explode"
```

Comment:
Ah I see why you want to use `__str__` now because user's function name is different from internal Snowflake function name (explode vs flatten). Maybe we should consider adding a field that override the name in error message? E.g. `user_visible_name` or something similar.

Reply:
That is a good suggestion. I'll add it to this PR.